File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/ServerTypes/Tanimlamalar/DrugsForm.ts
import { StringEditor, BooleanEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface DrugsForm {
    Name: StringEditor;
    IsActive: BooleanEditor;
    Barcode: StringEditor;
    CompanyName: StringEditor;
    AtcCode: StringEditor;
    AtcName: StringEditor;
}

export class DrugsForm extends PrefixedContext {
    static readonly formKey = 'Tanimlamalar.Drugs';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!DrugsForm.init)  {
            DrugsForm.init = true;

            var w0 = StringEditor;
            var w1 = BooleanEditor;

            initFormType(DrugsForm, [
                'Name', w0,
                'IsActive', w1,
                'Barcode', w0,
                'CompanyName', w0,
                'AtcCode', w0,
                'AtcName', w0
            ]);
        }
    }
}

File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/Tanimlamalar/Drugs/DrugsForm.cs
using Serenity.ComponentModel;

namespace MuayeneYonetimPortali.Tanimlamalar.Forms;

[FormScript("Tanimlamalar.Drugs")]
[BasedOnRow(typeof(DrugsRow), CheckNames = true)]
public class DrugsForm
{
    public string Name { get; set; }
    public bool IsActive { get; set; }
    public string Barcode { get; set; }
    public string CompanyName { get; set; }
    public string AtcCode { get; set; }
    public string AtcName { get; set; }
}